app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -0,0 +1,493 @@
+    }
+
+    private fun showAboutDialog() {
+        val meta = appRepository.readMeta()
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.addToBackStack(null)
+        AboutDialog
+            .newInstance(meta.version, meta.subtitle, meta.title)
+            .show(transaction, AboutDialog.FRAGMENT_TAG)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.menu_item_about -> {
+                showAboutDialog()
+                return true
+            }
+            R.id.menu_item_alarms -> {
+                AlarmList.startForResult(this)
+                return true
+            }
+            R.id.menu_item_settings -> {
+                SettingsActivity.startForResult(this)
+                return true
+            }
+            R.id.menu_item_schedule_changes -> {
+                openSessionChanges()
+                return true
+            }
+            R.id.menu_item_favorites -> {
+                openFavorites()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun openSessionDetails(sessionId: String) {
+        if (appRepository.updateSelectedSessionId(sessionId)) {
+            val sidePaneView = findViewById<FragmentContainerView>(R.id.detail)
+            if (sidePaneView == null) {
+                SessionDetailsActivity.startForResult(this)
+            } else {
+                SessionDetailsFragment.replaceAtBackStack(supportFragmentManager, R.id.detail, true)
+            }
+        }
+    }
+
+    override fun onSidePaneClose(fragmentTag: String) {
+        findViewById<View>(R.id.detail)?.let {
+            it.isVisible = false
+        }
+        if (fragmentTag == StarredListFragment.FRAGMENT_TAG) {
+            isFavoritesInSidePane = false
+        }
+        removeFragment(fragmentTag)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
+        super.onActivityResult(requestCode, resultCode, intent)
+        when (requestCode) {
+
+            AlarmList.REQUEST_CODE,
+            SessionDetailsActivity.REQUEST_CODE ->
+                if (resultCode == RESULT_CANCELED) {
+                    shouldScrollToCurrent = false
+                }
+
+            SettingsActivity.REQUEST_CODE ->
+                if (resultCode == RESULT_OK && intent != null) {
+                    val isAlternativeHighlightingUpdated = intent.getBooleanExtra(
+                        BUNDLE_KEY_ALTERNATIVE_HIGHLIGHTING_UPDATED, false)
+                    val isUseDeviceTimeZoneUpdated = intent.getBooleanExtra(
+                        BUNDLE_KEY_USE_DEVICE_TIME_ZONE_UPDATED, false)
+
+                    if (isAlternativeHighlightingUpdated || isUseDeviceTimeZoneUpdated) {
+                        if (findViewById<View>(R.id.schedule) != null && findFragment(FahrplanFragment.FRAGMENT_TAG) == null) {
```

Comment:
`findViewById<View>(R.id.schedule) is View` && findFragment(FahrplanFragment.FRAGMENT_TAG) !is Fragment`